catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java

```diff
@@ -103,22 +89,22 @@ private EntitySnapshotObj.Builder importIcebergTable(
     // snapshot!=null means that we already have the snapshot (via a committing operation - like a
     // table update) and do not need to import it but can just store it.
     if (snapshot == null) {
-      IcebergTableMetadata tableMetadata;
       StorageUri metadataLocation = StorageUri.of(content.getMetadataLocation());
+      NessieTable table;
+      IcebergTableMetadata tableMetadata;
       try {
         InputStream input = taskRequest.objectIO().readObject(metadataLocation);
         if (metadataLocation.requiredPath().endsWith(".gz")
             || metadataLocation.requiredPath().endsWith(".gz.metadata.json")) {
           input = new GZIPInputStream(input);
         }
         tableMetadata = IcebergJson.objectMapper().readValue(input, IcebergTableMetadata.class);
-      } catch (IOException e) {
-        throw new IOException(
+        table = entityObjForContent(content, tableMetadata, entityObjId);
+      } catch (Exception e) {
```

Comment:
Not sure about this (would exclude the location from the exception message), but as a suggestion.
```suggestion
      } catch (RuntimeException e) {
        throw e;
      } catch (Exception e) {
```

Reply:
If we do not report the file path to the client, the error on the client side looks kind of cryptic... something like `Forbidden: java.io.IOException: software.amazon.awssdk.services.s3.model.S3Exception: Access Denied. (Service: S3, Status Code: 403, Request ID: )`

With this change we can report it as `Forbidden: Failed to read table metadata from s3://bucket1/warehouse/ns/table1_06a7ebe4-7dc6-4526-a85e-578993e09a12_test_access_denied_file (due to: software.amazon.awssdk.services.s3.model.S3Exception: Access Denied. (Service: S3, Status Code: 403, Request ID: ))`.

I know S3 servers do not report paths in errors, however, in this case my rationale is that the user is in control of both  the credentials and the S3 objects. Nessie validates the user's credentials to access the catalog before we produce this error... so I reasoned that it should be fine to expose the extra info for ease of trouble-shooting. WDYT?

Reply:
SGTM